frame/support/test/tests/pallet_ui/pass/dev_mode_valid.rs

```diff
@@ -42,4 +48,71 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {}
 }
 
-fn main() {}
+impl frame_system::Config for Runtime {
+	type BaseCallFilter = frame_support::traits::Everything;
+	type RuntimeOrigin = RuntimeOrigin;
+	type Index = u64;
+	type BlockNumber = u32;
+	type RuntimeCall = RuntimeCall;
+	type Hash = sp_runtime::testing::H256;
+	type Hashing = sp_runtime::traits::BlakeTwo256;
+	type AccountId = u64;
+	type Lookup = sp_runtime::traits::IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type RuntimeEvent = RuntimeEvent;
+	type BlockHashCount = ConstU32<250>;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = ();
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
+	type MaxConsumers = ConstU32<16>;
+}
+
+pub type Header = sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>;
+pub type Block = sp_runtime::generic::Block<Header, UncheckedExtrinsic>;
+pub type UncheckedExtrinsic = sp_runtime::generic::UncheckedExtrinsic<u32, RuntimeCall, (), ()>;
+
+frame_support::construct_runtime!(
+	pub struct Runtime where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic
+	{
+		// Exclude part `Storage` in order not to check its metadata in tests.
+		System: frame_system exclude_parts { Pallet, Storage },
+		Example: pallet,
+	}
+);
+
+impl pallet::Config for Runtime {
+	
+}
+
+fn main() {
+	use frame_support::{pallet_prelude::*};
+	use storage::unhashed;
+	use sp_io::{
+		hashing::{blake2_128, twox_128},
+		TestExternalities,
+	};
+
+	fn blake2_128_concat(d: &[u8]) -> Vec<u8> {
+		let mut v = blake2_128(d).to_vec();
+		v.extend_from_slice(d);
+		v
+	}
+
+	TestExternalities::default().execute_with(|| {
+		pallet::MyStorageMap::<Runtime>::insert(1, 2);
+		let mut k = [twox_128(b"Example"), twox_128(b"MyStorageMap")].concat();
+		k.extend(1u32.using_encoded(blake2_128_concat));
+		assert_eq!(unhashed::get::<u64>(&k), Some(2u64));
```

Comment:
Yes thanks for the test!  

Normally we only test UI behaviour in the UI tests and not logic, but i think it is fine.